Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -716,6 +716,32 @@ theorem eq_zero_of_comapDomain_eq_zero [AddCommMonoid M] (f : α → β) (l : β
   exact h b (hb.2.symm ▸ ha)
 #align finsupp.eq_zero_of_comap_domain_eq_zero Finsupp.eq_zero_of_comapDomain_eq_zero
 
+lemma exists_mem_embDomain_eq [Zero M] {f : α ↪ β} {g : β →₀ M} :
+    (∃ (v : α →₀ M), embDomain f v = g) ↔ (g.support : Set β) ⊆ (Set.range f) := by
+  constructor
+  · rintro ⟨v, rfl⟩
+    simp only [Finsupp.support_embDomain, coe_map, Set.image_subset_iff,
+      Set.preimage_range, Set.subset_univ]
+  · intro hg
+    use Finsupp.comapDomain f g (f.injective.injOn _)
+    ext b
```

Comment:
Is there a lemma you can use here about `embDomain _ (comapDomain _ _)`? If not, can you write that lemma?

Reply:
The only existing way would rewrite `embDomain` as `mapDomain`, but this assumes `AddCommMonoid M`, which is not needed.
I'll write that lemma.

Reply:
(mapDomain sums over preimages, but when f is injective, the sum is reduced to a subsingleton).
On the other hand, docs#mapDomain_comapDomain makes an injectivity assumption; my guess is that it should be rephrased as  embDomain_comapDomain, and the initial mapDomain_comapDomain proved in terms of it.